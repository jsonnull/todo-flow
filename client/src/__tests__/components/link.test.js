import React from 'react';
import renderer from 'react-test-renderer';

import Link from '../../components/link';

describe('Link', () => {
  let tree: Object;
  it('renders without crashing', () => {
    const component = renderer.create(<Link />);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a span when active is true', () => {
    const component = renderer.create(<Link active />);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick() on click', () => {
    const onClick = jest.fn();
    const component = renderer.create(<Link onClick={onClick} />);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick({ preventDefault: () => {} });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });
});
