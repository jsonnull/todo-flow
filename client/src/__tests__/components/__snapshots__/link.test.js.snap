exports[`Link calls onClick() on click 1`] = `
<a
  href="#"
  onClick={[Function]} />
`;

exports[`Link calls onClick() on click 2`] = `
<a
  href="#"
  onClick={[Function]} />
`;

exports[`Link renders a span when active is true 1`] = `<span />`;

exports[`Link renders without crashing 1`] = `
<a
  href="#"
  onClick={[Function]} />
`;
